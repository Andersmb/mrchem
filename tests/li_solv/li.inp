# vim:syntax=sh:
world_prec = 1.0e-3               # Overall relative precision
world_size = 5                    # Size of simulation box 2^n

MPI {
  numerically_exact = true        # Guarantee identical results in MPI
}

Molecule {
  charge = 1
$coords
Li   0.0     0.0    0.0
$end
}

WaveFunction {
  method = pbe0                     # Wave function method (HF or DFT)
}

Environment {
  run_environment = true
  algorithm = scrf
  kain = 5
  max_iter = 100
  convergence_criterion = dynamic
  extrapolate_Vr = true

  Cavity {
    cavity_width = 0.5
$spheres
0.0  0.0  0.0  4.0
$end
  }

  Permittivity {
    epsilon_in = 1.0
    epsilon_out = 2.0
    formulation = exponential
  }
}

SCF {
  run = true
  kain = 5
  orbital_thrs = 1.0e-1
  guess_type = sad_dz
}
