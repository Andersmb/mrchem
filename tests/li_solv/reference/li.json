{
  "input": {
    "molecule": {
      "cavity_coords": [
        {
          "center": [
            0.0,
            0.0,
            0.0
          ],
          "radius": 4.0
        },
        {
          "center": [
            0.0,
            0.0,
            0.0
          ],
          "radius": 4.0
        }
      ],
      "cavity_width": 0.5,
      "charge": 1,
      "coords": [
        {
          "atom": "li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 1
    },
    "mpi": {
      "bank_size": -1,
      "numerically_exact": true,
      "shared_memory_size": 10000
    },
    "mra": {
      "basis_order": 5,
      "basis_type": "interpolating",
      "boxes": [
        2,
        2,
        2
      ],
      "corner": [
        -1,
        -1,
        -1
      ],
      "max_scale": 20,
      "min_scale": -4
    },
    "printer": {
      "file_name": "li.inp",
      "print_level": 0,
      "print_mpi": false,
      "print_prec": 6,
      "print_width": 75
    },
    "rsp_calculations": {},
    "scf_calculation": {
      "fock_operator": {
        "coulomb_operator": {
          "poisson_prec": 0.001,
          "shared_memory": false
        },
        "exchange_operator": {
          "exchange_prec": -1.0,
          "poisson_prec": 0.001
        },
        "kinetic_operator": {
          "derivative": "abgv_55"
        },
        "nuclear_operator": {
          "proj_prec": 0.001,
          "shared_memory": false,
          "smooth_prec": 0.001
        },
        "reaction_operator": {
          "accelerate_Vr": true,
          "algorithm": "scrf",
          "convergence_criterion": "dynamic",
          "density_type": "total",
          "epsilon_in": 1.0,
          "epsilon_out": 2.0,
          "formulation": "exponential",
          "kain": 5,
          "max_iter": 100,
          "poisson_prec": 0.001
        },
        "xc_operator": {
          "shared_memory": false,
          "xc_functional": {
            "cutoff": 0.0,
            "functionals": [
              {
                "coef": 1.0,
                "name": "pbe0"
              }
            ],
            "spin": false
          }
        }
      },
      "initial_guess": {
        "file_basis": "initial_guess/mrchem.bas",
        "file_chk": "checkpoint/phi_scf",
        "file_gto_a": "initial_guess/mrchem.moa",
        "file_gto_b": "initial_guess/mrchem.mob",
        "file_gto_p": "initial_guess/mrchem.mop",
        "file_phi_a": "initial_guess/phi_a_scf",
        "file_phi_b": "initial_guess/phi_b_scf",
        "file_phi_p": "initial_guess/phi_p_scf",
        "localize": false,
        "method": "DFT (PBE0)",
        "prec": 0.001,
        "restricted": true,
        "type": "sad",
        "zeta": 2
      },
      "properties": {
        "dipole_moment": {
          "dip-1": {
            "operator": "h_e_dip",
            "precision": 0.001,
            "r_O": [
              0.0,
              0.0,
              0.0
            ]
          }
        }
      },
      "scf_solver": {
        "checkpoint": false,
        "energy_thrs": -1.0,
        "file_chk": "checkpoint/phi_scf",
        "final_prec": 0.001,
        "helmholtz_prec": -1.0,
        "kain": 5,
        "localize": false,
        "max_iter": 100,
        "method": "DFT (PBE0)",
        "orbital_thrs": 0.1,
        "rotation": 0,
        "start_prec": 0.001
      }
    },
    "schema_name": "mrchem_input",
    "schema_version": 1
  },
  "output": {
    "properties": {
      "center_of_mass": [
        0.0,
        0.0,
        0.0
      ],
      "charge": 1,
      "dipole_moment": {
        "dip-1": {
          "magnitude": 7.166178513585691e-14,
          "r_O": [
            0.0,
            0.0,
            0.0
          ],
          "vector": [
            0.0,
            0.0,
            0.0
          ],
          "vector_el": [
            0.0,
            0.0,
            0.0
          ],
          "vector_nuc": [
            0.0,
            0.0,
            0.0
          ]
        }
      },
      "geometry": [
        {
          "symbol": "Li",
          "xyz": [
            0.0,
            0.0,
            0.0
          ]
        }
      ],
      "multiplicity": 1,
      "orbital_energies": {
        "energy": [
          -2.2187679061661942
        ],
        "occupation": [
          2.0
        ],
        "spin": [
          "p"
        ],
        "sum_occupied": -4.4375358123323885
      },
      "scf_energy": {
        "E_ee": 3.354626053968655,
        "E_eext": 0.0,
        "E_el": -7.134932723751487,
        "E_en": -16.47793719577992,
        "E_kin": 7.572169190897237,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": -0.18173935351782744,
        "E_tot": -7.316672077269315,
        "E_x": -0.4192129418515808,
        "E_xc": -1.2857442250812878,
        "Er_el": 0.12116639409540789,
        "Er_nuc": -0.18173935351782744,
        "Er_tot": -0.06057295942241949
      }
    },
    "provenance": {
      "creator": "MRChem",
      "mpi_processes": 1,
      "nthreads": 4,
      "routine": "mrchem.x",
      "total_cores": 4,
      "version": "1.1.0-alpha"
    },
    "rsp_calculations": null,
    "scf_calculation": {
      "initial_energy": {
        "E_ee": 3.33279898519371,
        "E_eext": 0.0,
        "E_el": -7.2226490727778625,
        "E_en": -16.530753867922876,
        "E_kin": 7.675563119477212,
        "E_next": 0.0,
        "E_nn": 0.0,
        "E_nuc": 0.0,
        "E_tot": -7.2226490727778625,
        "E_x": -0.41648444551024355,
        "E_xc": -1.2837728640156645,
        "Er_el": 0.0,
        "Er_nuc": 0.0,
        "Er_tot": 0.0
      },
      "scf_solver": {
        "converged": true,
        "cycles": [
          {
            "energy_terms": {
              "E_ee": 3.354626053968655,
              "E_eext": 0.0,
              "E_el": -7.134932723751487,
              "E_en": -16.47793719577992,
              "E_kin": 7.572169190897237,
              "E_next": 0.0,
              "E_nn": 0.0,
              "E_nuc": -0.18173935351782744,
              "E_tot": -7.316672077269315,
              "E_x": -0.4192129418515808,
              "E_xc": -1.2857442250812878,
              "Er_el": 0.12116639409540789,
              "Er_nuc": -0.18173935351782744,
              "Er_tot": -0.06057295942241949
            },
            "energy_total": -7.316672077269315,
            "energy_update": 0.09402300449145251,
            "mo_residual": 0.060097618192063045,
            "wall_time": 8.343238766
          }
        ],
        "wall_time": 8.343593713
      },
      "success": true
    },
    "schema_name": "mrchem_output",
    "schema_version": 1,
    "success": true
  }
}
