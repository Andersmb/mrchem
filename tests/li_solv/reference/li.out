                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                                                                     ***
***          __  __ ____   ____ _                                       ***
***         |  \/  |  _ \ / ___| |__   ___ _ __ ___                     ***
***         | |\/| | |_) | |   | '_ \ / _ \ '_ ` _ \                    ***
***         | |  | |  _ <| |___| | | |  __/ | | | | |                   ***
***         |_|  |_|_| \_\\____|_| |_|\___|_| |_| |_|                   ***
***                                                                     ***
***         VERSION            1.1.0-alpha                              ***
***                                                                     ***
***         Git branch         fastDFT                                  ***
***         Git commit hash    243010639c3fcaa38b60-dirty               ***
***         Git commit author  gitpeterwind                             ***
***         Git commit date    Thu Mar 18 12:04:47 2021 +0100           ***
***                                                                     ***
***         Contact: luca.frediani@uit.no                               ***
***                                                                     ***
***         Radovan Bast            Magnar Bjorgve                      ***
***         Roberto Di Remigio      Antoine Durdek                      ***
***         Luca Frediani           Gabriel Gerez                       ***
***         Stig Rune Jensen        Jonas Juselius                      ***
***         Rune Monstad            Peter Wind                          ***
***                                                                     ***
***************************************************************************

---------------------------------------------------------------------------

 MPI processes           :       (no bank)                               1
 OpenMP threads          :                                               4
 Total cores             :                                               4
                                                                           
---------------------------------------------------------------------------

XCFun DFT library Copyright 2009-2020 Ulf Ekstrom and contributors.
See http://dftlibs.org/xcfun/ for more information.

This is free software; see the source code for copying conditions.
There is ABSOLUTELY NO WARRANTY; not even for MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE. For details see the documentation.
Scientific users of this library should cite
U. Ekstrom, L. Visscher, R. Bast, A. J. Thorvaldsen and K. Ruud;
J.Chem.Theor.Comp. 2010, DOI: 10.1021/ct100117s

---------------------------------------------------------------------------

 MRCPP version         : 1.4.0-alpha
 Git branch            : HEAD
 Git commit hash       : d498e7969d8b9998a405-dirty
 Git commit author     : Stig Rune Jensen
 Git commit date       : Thu Mar 18 10:41:39 2021 +0100

 Linear algebra        : EIGEN v3.3.7
 Parallelization       : OpenMP (4 threads)

---------------------------------------------------------------------------


                                                                           
===========================================================================
                         MultiResolution Analysis
---------------------------------------------------------------------------
 polynomial order      : 5
 polynomial type       : Interpolating
---------------------------------------------------------------------------
 total boxes           : 8
 boxes                 : [          2           2           2 ]
 unit lengths          : [   16.00000    16.00000    16.00000 ]
 scaling factor        : [    1.00000     1.00000     1.00000 ]
 lower bounds          : [  -16.00000   -16.00000   -16.00000 ]
 upper bounds          : [   16.00000    16.00000    16.00000 ]
 total length          : [   32.00000    32.00000    32.00000 ]
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                        Initializing Molecule                        ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               1
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                 Computing Ground State Wavefunction                 ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial orbitals
 Method                  : Diagonalize SAD Hamiltonian
 Precision               : 1.00000e-03
 Restricted              : True
 Functional              : LDA (SVWN5)
 AO basis                : Hydrogenic orbitals
 Zeta quality            : 2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                            Molecular Orbitals
---------------------------------------------------------------------------
 Alpha electrons         :                                               1
 Beta electrons          :                                               1
 Total electrons         :                                               2
---------------------------------------------------------------------------
    n  Occ Spin          :                                            Norm
---------------------------------------------------------------------------
    0    2    p          :                              1.000000000000e+00
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Compute initial energy
 Method                  : DFT (PBE0)
 Precision               : 1.00000e-03
 Localization            : Off
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
                        Molecular Energy (initial)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.675563119477
 E-N energy              :            (au)                -16.530753867923
 Coulomb energy          :            (au)                  3.332798985194
 Exchange energy         :            (au)                 -0.416484445510
 X-C energy              :            (au)                 -1.283772864016
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                  0.000000000000
 El. Reac. Energy        :            (au)                  0.000000000000
 Nuc. Reac. Energy       :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.222649072778
 Nuclear energy          :            (au)                  0.000000000000
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.222649072778e+00
                         :      (kcal/mol)             -4.532280684432e+03
                         :        (kJ/mol)             -1.896306239597e+04
                         :            (eV)             -1.965382850004e+02
===========================================================================


===========================================================================
                        Orbital Energies (initial)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -2.332093284752
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.664186569505
===========================================================================


~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Calculation             : Optimize ground state orbitals
 Method                  : DFT (PBE0)
 Checkpointing           : Off
 Max iterations          : 100
 KAIN solver             : 5
 Localization            : Off
 Diagonalization         : First two iterations
 Start precision         : 1.00000e-03
 Final precision         : 1.00000e-03
 Helmholtz precision     : Dynamic
 Energy threshold        : Off
 Orbital threshold       : 1.00000e-01
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


===========================================================================
 Iter           MO residual             Total energy                Update
---------------------------------------------------------------------------
    0          1.000000e+00          -7.222649072778         -7.222649e+00
    1          6.009762e-02          -7.316672077269         -9.402300e-02
---------------------------------------------------------------------------
                      SCF converged in 1 iterations!
===========================================================================


                                                                           
***************************************************************************
***                                                                     ***
***                    Printing Molecular Properties                    ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
===========================================================================
                                 Molecule
---------------------------------------------------------------------------
 Charge                  :                                               1
 Multiplicity            :                                               1
---------------------------------------------------------------------------
    N    Atom            :               x               y               z
---------------------------------------------------------------------------
    0      Li            :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Center of mass          :        0.000000        0.000000        0.000000
===========================================================================


===========================================================================
                         Molecular Energy (final)
---------------------------------------------------------------------------
 Kinetic energy          :            (au)                  7.572169190897
 E-N energy              :            (au)                -16.477937195780
 Coulomb energy          :            (au)                  3.354626053969
 Exchange energy         :            (au)                 -0.419212941852
 X-C energy              :            (au)                 -1.285744225081
 Ext. field (el)         :            (au)                  0.000000000000
---------------------------------------------------------------------------
 N-N energy              :            (au)                  0.000000000000
 Ext. field (nuc)        :            (au)                  0.000000000000
 Tot. Reac. Energy       :            (au)                 -0.060572959422
 El. Reac. Energy        :            (au)                  0.121166394095
 Nuc. Reac. Energy       :            (au)                 -0.181739353518
---------------------------------------------------------------------------
 Electronic energy       :            (au)                 -7.134932723751
 Nuclear energy          :            (au)                 -0.181739353518
---------------------------------------------------------------------------
 Total energy            :            (au)             -7.316672077269e+00
                         :      (kcal/mol)             -4.591281010054e+03
                         :        (kJ/mol)             -1.920991975854e+04
                         :            (eV)             -1.990967811792e+02
===========================================================================


===========================================================================
                         Orbital Energies (final)
---------------------------------------------------------------------------
    n  Occ Spin          :                                         Epsilon
---------------------------------------------------------------------------
    0    2    p          :            (au)                 -2.218767906166
---------------------------------------------------------------------------
 Sum occupied            :            (au)                 -4.437535812332
===========================================================================


===========================================================================
                           Dipole Moment (dip-1)
---------------------------------------------------------------------------
 r_O                     :        0.000000        0.000000        0.000000
---------------------------------------------------------------------------
 Electronic vector       :       -0.000000       -0.000000       -0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Nuclear vector          :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
---------------------------------------------------------------------------
 Total vector            :        0.000000        0.000000        0.000000
 Magnitude               :            (au)                        0.000000
                         :         (Debye)                        0.000000
===========================================================================


                                                                           
                                                                           
***************************************************************************
***                                                                     ***
***                            Exiting MRChem                           ***
***                                                                     ***
***                       Wall time :  0h  0m 33s                       ***
***                                                                     ***
***************************************************************************
                                                                           
                                                                           
